src/cloud-api-adaptor/test/e2e/assessment_helpers.go

```diff
@@ -708,12 +708,34 @@ func DeleteAndWaitForNamespace(ctx context.Context, client klient.Client, namesp
 	return nil
 }
 
-func AddImagePullSecretToDefaultServiceAccount(ctx context.Context, client klient.Client, secretName string) error {
+func getDefaultServiceAccount(ctx context.Context, client klient.Client) (*v1.ServiceAccount, error) {
 	clientSet, err := kubernetes.NewForConfig(client.RESTConfig())
 	if err != nil {
-		return err
+		return nil, err
 	}
 	serviceAccount, err := clientSet.CoreV1().ServiceAccounts(E2eNamespace).Get(context.TODO(), "default", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return serviceAccount, nil
+}
+
+func setImagePullSecretsOnServiceAccount(ctx context.Context, client klient.Client, serviceAccount *v1.ServiceAccount, imagePullSecrets []v1.LocalObjectReference) error {
+	clientSet, err := kubernetes.NewForConfig(client.RESTConfig())
+	if err != nil {
+		return err
+	}
+	serviceAccount.ImagePullSecrets = imagePullSecrets
+	_, err = clientSet.CoreV1().ServiceAccounts(E2eNamespace).Update(context.TODO(), serviceAccount, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+	log.Infof("ServiceAccount %s updated successfully.", "default")
```

Comment:
as this fn is generic, I would move the log statement to the caller of the fn (`Add...ToDefaultServiceAccount`)

we don't need string templating if we use 2 constant strings.